File: _BY_MONTH/_Feb/Thesis-physics-master/browser/engine/sphere.js
/**
 * @fileoverview Sphere file.
 * @author sebastian.poreba@gmail.com (Sebastian Poreba)
 */


goog.provide('smash.Sphere');



/**
 * @struct
 * @constructor
 */
smash.Sphere = function() {
  /**
   * @type {number}
   */
  this.positionX = 0.1;

  /**
   * @type {number}
   */
  this.positionY = 0.1;

  /**
   * @type {number}
   */
  this.positionZ = 0.1;

  /**
   * @type {number}
   */
  this.velocityX = 0.1;

  /**
   * @type {number}
   */
  this.velocityY = 0.1;

  /**
   * @type {number}
   */
  this.velocityZ = 0.1;

  /**
   * @type {number}
   */
  this.radius = 5.5;

  /**
   * @type {number}
   */
  this.mass = 3.5;
};


/**
 * @param {number} stepTime
 */
smash.Sphere.prototype.step = function(stepTime) {
  this.positionX += this.velocityX * stepTime;
  this.positionY += this.velocityY * stepTime;
  this.positionZ += this.velocityZ * stepTime;
};

File: _BY_MONTH/_Feb/Thesis-physics-master/doc/spheres/sphere.cpp
/**
 * @fileoverview Sphere file.
 * @author sebastian.poreba@gmail.com (Sebastian Poreba)
 */

#include "sphere.h"

smash::Sphere::Sphere() {
  positionX = 0;
  positionY = 0;
  positionZ = 0;
  velocityX = 0;
  velocityY = 0;
  velocityZ = 0;
  radius = 5.5;
  mass = 3.5;
};

/**
 * @param stepTime
 */
void smash::Sphere::step(float stepTime) {
  this->positionX += this->velocityX * stepTime;
  this->positionY += this->velocityY * stepTime;
  this->positionZ += this->velocityZ * stepTime;
};

